src/tests/ftest/deployment/ior_per_rank.py

```diff
@@ -0,0 +1,130 @@
+                if good_node not in self.good_nodes:
+                    self.good_nodes.append(good_node)
+            except TestFail as _error:
+                # collect bad nodes
+                failed_node = self.server_managers[0].get_host(rank)
+                if failed_node not in self.failed_nodes:
+                    self.failed_nodes[failed_node] = [rank]
+                else:
+                    self.failed_nodes[failed_node].append(rank)
+                if failed_node in self.good_nodes:
+                    self.good_nodes.remove(failed_node)
+        # Destroy pool, to be sure we use newly created pool in next iteration
+        self.pool.destroy()
+        self.pool = None
+
+    def test_iorperrank(self):
+        """
+        Test Description: Test to check node health using ior with daos.
+                          Start entire daos system on each rack of a group.
+                          Create pool for each rank.
+                          Run ior using 4 good clients over each pool.
+                              Large transfer size: 1M
+                              Small transfer size: 256B
+                          Compare results and isolate bad nodes.
+        :avocado: tags=all,deployment,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=ior
+        :avocado: tags=iorperrank
+        """
+
+        # test params
+        self.failed_nodes = {}
+        self.good_nodes = []
+        self.transfer_sizes = self.params.get("transfer_sizes", self.ior_cmd.namespace)
+
+        # Write/Read performance thresholds and expectations
+        self.write_x = self.params.get("write_x", self.ior_cmd.namespace, None)
+        self.read_x = self.params.get("read_x", self.ior_cmd.namespace, None)
+        self.expected_bw = self.params.get("expected_bw", self.ior_cmd.namespace, None)
+        self.expected_iops = self.params.get("expected_iops", self.ior_cmd.namespace, None)
+
+        if (self.write_x is None or self.read_x is None or self.expected_bw is None or
+                self.expected_iops is None):
+            self.fail("Failed to get write_x, read_x, expected_bw, expected_iops from config")
+
+        # create a list of all the ranks
+        rank_list = self.server_managers[0].get_host_ranks(self.hostlist_servers)
+        self.log.info("rank_list: %s", rank_list)
+
+        # run ior over DFS
+        self.ior_cmd.api.update("DFS")
+        for rank in rank_list:
+            self.execute_ior_thread(rank)
+
+        # list of good nodes
+        if self.good_nodes:
+            self.log.info("List of good nodes: %s", self.good_nodes)
+
+        # list the failed node and the rank number associated with that node
+        if self.failed_nodes:
+            self.log.info("List of failed ranks with corresponding nodes")
+            for node, rank in self.failed_nodes.items():
+                self.log.info("Node: %s, Rank: %s", node, rank)
+            self.fail()
```

Comment:
We should add a message to `fail` so it's easy to identify failures. Maybe something like?
```suggestion
            self.fail("Performance check failed for one or more nodes")
```

Reply:
Done